build-system/tasks/release/index.js

```diff
@@ -0,0 +1,375 @@
+  });
+
+  logSeparator_();
+
+  return distFlavors;
+}
+
+async function compileDistFlavors_(distFlavors, outputTempDir) {
+  for (const {flavorType, command} of distFlavors) {
+    log('Compiling flavor', `${green(flavorType)}:`);
+
+    execOrDie('gulp clean');
+    execOrDie(command);
+
+    const flavorTempDistDir = path.join(outputTempDir, flavorType);
+    log('Moving build artifacts to', `${cyan(flavorTempDistDir)}...`);
+    await fs.ensureDir(flavorTempDistDir);
+
+    await Promise.all(
+      DIST_DIRS.map((distDir) =>
+        fs.move(distDir, path.join(flavorTempDistDir, distDir))
+      )
+    );
+
+    log('Copying static files...');
+    await Promise.all([
+      // Directory-to-directory copy from the ./static sub-directory.
+      fs.copy(STATIC_FILES_DIR, path.join(flavorTempDistDir, 'dist')),
+      // Individual files to copy from the Git repository.
+      ...STATIC_FILE_PATHS.map((staticFilePath) =>
+        fs.copy(
+          staticFilePath,
+          path.join(flavorTempDistDir, 'dist', path.basename(staticFilePath))
+        )
+      ),
+      // Individual files to copy from the resulting build artifacts.
+      ...Object.entries(POST_BUILD_MOVES).map(([from, to]) =>
+        fs.copy(
+          path.join(flavorTempDistDir, from),
+          path.join(flavorTempDistDir, to)
+        )
+      ),
+    ]);
+
+    logSeparator_();
+  }
+}
+
+async function fetchAmpSw_(distFlavorTypes, outputTempDir) {
+  for (const flavorType of distFlavorTypes) {
+    await fs.ensureDir(path.join(outputTempDir, flavorType, 'dist/sw'));
+  }
+
+  const ampSwBaseTempDir = path.join(outputTempDir, 'base/dist/sw');
+
+  const ampSwTarballUrl = await fetch(
+    'https://registry.npmjs.org/@ampproject/amp-sw'
+  )
+    .then((res) => res.json())
+    .then((json) => {
+      const {latest} = json['dist-tags'];
+      return json.versions[latest].dist.tarball;
+    });
+  await fetch(ampSwTarballUrl).then(
+    (res) =>
+      new Promise((resolve) => {
+        const tarWritableStream = tar.extract({
+          cwd: ampSwBaseTempDir,
+          filter: (path) => path.startsWith('package/dist'),
+          strip: 2, // to strip "package/dist/".
+        });
+        res.body.pipe(tarWritableStream);
+        tarWritableStream.on('end', resolve);
+      })
+  );
+
+  await Promise.all(
+    distFlavorTypes
+      .filter((flavorType) => flavorType != 'base')
+      .map((flavorType) =>
+        fs.copy(
+          ampSwBaseTempDir,
+          path.join(outputTempDir, flavorType, 'dist/sw')
+        )
+      )
+  );
+
+  logSeparator_();
+}
+
+async function populateOrgCdn_(distFlavors, outputTempDir, outputDir) {
+  for (const {flavorType, rtvPrefixes} of distFlavors) {
+    await Promise.all(
+      rtvPrefixes.map((rtvPrefix) => {
+        const rtvNumber = `${rtvPrefix}${VERSION}`;
+        const rtvPath = path.join(outputDir, 'org-cdn/rtv', rtvNumber);
+        return fs
+          .ensureDir(rtvPath)
+          .then(() =>
```

Comment:
Done. Since this is part of the `map` callback, which is inside the `Promise.all` call, I changed it to:
```js
await Promise.all(rtvPrefixes.map(async (rtvPrefix) => {
  ...
  await fs.ensureDir(rtvPath);
  return fs.copy(path.join(outputTempDir, flavorType, 'dist'), rtvPath);
})
```

So the map returns an array of promises, and the containing function awaits that all of them are done. This is because `fs` actions are capable of running in parallel so might as well let them